x-pack/plugins/endpoint/public/applications/endpoint/lib/location/is_on_page.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EndpointAppLocation } from '../../types';
+
+let currentLocation: EndpointAppLocation;
+let fromLocation: EndpointAppLocation;
+
+export const storeCurrentLocation = (loc: EndpointAppLocation) => {
+  fromLocation = currentLocation;
+  currentLocation = loc;
+};
+
+export const isOnPage = (pageRoute: EndpointAppLocation['pathname']) => {
+  if (!currentLocation) {
+    throw new Error('EndpointAppLocation has not been captured yet');
+  }
+  return currentLocation.pathname === pageRoute;
+};
+
+export const isOnDetailsPage = (pageBaseRoute: EndpointAppLocation['pathname']) => {
+  if (!currentLocation) {
+    throw new Error('EndpointAppLocation has not been captured yet');
+  }
+  const pathnameParts = currentLocation.pathname.split('/');
```

Comment:
I was hoping we did not have to parse the pathname to get information about the route and would instead use react-router's `match` object to determine this (ex. `match.path ==== '/policy/:id'`). This object also gives us a parsed url params, which we should also start to save along with the `EndpointAppLocation` object - since that is needed by middleware as well.

The challenge with that is that object (`match`) only gets set at the `<Route>` level - in `RouteCapture`, because it's included at the root level, the `match.path` is always `/` which is not what we want. I'm thinking that we should replace the use of `<Route .../>` with a custom HOC (e.x. `<EndpointRoute .../>` that can do the work that `<RouteCapture>` is doing today ++ also capture and store the `match` object which includes the route that was matched.

let me know if you have any question about anything above :) 

_(ps. I also think that the router `path`'s could be centralized as "constants" so that we centralize those)_

Reply:
@paul-tavares is the `match` object you're referring to come from the `useRouteMatch()` hook?

Also, are you thinking we'd access the hook in the individual HOC's and then store the `match` object from the hook?

Reply:
@kevinlog no - that hook needs to be fed the route `path` and then it will try to match that path against the current URL and returns the match (which might not have matched :) ).

What I was thinking about is using the `match` object that is automatically given to React-Router's components in its various places (see: https://reacttraining.com/react-router/web/api/match).

I would envision something like this (maybe 🤔 ): 

```jsx
const EndpointRoute = ({path, component: ViewComponent}) => {
  return <Route path={path} render={(match, location) => {
     // the `render()` prop is given the `routeProps`, so match and location here for reference
    
    // Store the match and location in global
    // Maybe still fire a store action ?

    // render component for this route
    return <ViewComponent />
  }} />
}
```

Just a quick sample of what was in my mind. This should be improved (maybe) to support as much as possible of the normal behaviour of `<Route>`

Reply:
@paul-tavares 
Just a quick update here, it looks like `<Switch>` needs all children to be `<Route>` elements. https://reacttraining.com/react-router/web/api/Switch

The HOC's here is causing some odd behavior.  It will render the first HOC in the set without issues, but any other HOC containing a `<Route>` won't be found/not rendered.  I think we need to use `<Route>` directly in the `<Switch>` and perhaps rely on a `render()` method that we can pass around.

here's a quick and dirty sample to illustrate, let me know of any questions.

```
<Route
            path="/policy"
            exact
            render={({ match, location }) => {
              // dispatch({ type: 'userChangedUrl', payload: location });
              console.log('storeCurrentMatch')
              storeCurrentMatch(match);
              return <PolicyList />;
            }}
          />
```

Reply:
OHHHH 😞 

What about if we use an HOC around the component that the route uses?

Like: 

```jsx
const RouteViewPolicyList = memo(({match, location}) => {
  // store match, location
  return <PolicyList />
});
```
we can probably create a factory to build the above HOC instead of having to define the code to store the location/match in multiple places (ex. `withRouteCapture(renderFunctionHere)`)

With something like the above we can keep using `Route>`:

```jsx
<Switch>
  //...
  <Route path="/policy" exact component={RouteViewPolicyList} />
</Switch>
```